Compilation failure in YARP 3.5.100 with thrift structure of maps that contains lists

**Describe the bug**
In the project `blf`, we have the following thrift structure:
~~~
struct VectorsCollection
{
    1: map<string, list<double>> vectors;
}
~~~

However, the compilation of the corresponding generated code fails with the error:
~~~
2021-12-24T13:58:14.0840628Z [12/181] Building CXX object src/YarpUtilities/CMakeFiles/VectorsCollection.dir/autogenerated/src/VectorsCollection.cpp.o
2021-12-24T13:58:14.0842059Z FAILED: src/YarpUtilities/CMakeFiles/VectorsCollection.dir/autogenerated/src/VectorsCollection.cpp.o 
2021-12-24T13:58:14.0849257Z /usr/share/miniconda3/envs/test/bin/x86_64-conda-linux-gnu-c++ -DUSING_DEPRECATED_UPPERCASE_YARP_OS_TARGET -DVectorsCollection_EXPORTS -I/home/runner/work/robotology-superbuild/robotology-superbuild/build/src/bipedal-locomotion-framework/src/YarpUtilities/autogenerated/include -isystem /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/include -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /usr/share/miniconda3/envs/test/include -O3 -DNDEBUG -fPIC -MD -MT src/YarpUtilities/CMakeFiles/VectorsCollection.dir/autogenerated/src/VectorsCollection.cpp.o -MF src/YarpUtilities/CMakeFiles/VectorsCollection.dir/autogenerated/src/VectorsCollection.cpp.o.d -o src/YarpUtilities/CMakeFiles/VectorsCollection.dir/autogenerated/src/VectorsCollection.cpp.o -c /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/bipedal-locomotion-framework/src/YarpUtilities/autogenerated/src/VectorsCollection.cpp
2021-12-24T13:58:14.0856864Z /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/bipedal-locomotion-framework/src/YarpUtilities/autogenerated/src/VectorsCollection.cpp: In member function 'bool BipedalLocomotion::YarpUtilities::VectorsCollection::read_vectors(yarp::os::idl::WireReader&)':
2021-12-24T13:58:14.0860258Z /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/bipedal-locomotion-framework/src/YarpUtilities/autogenerated/src/VectorsCollection.cpp:99:18: error: redeclaration of 'uint32_t _size'
2021-12-24T13:58:14.0861621Z    99 |         uint32_t _size;
2021-12-24T13:58:14.0861961Z       |                  ^~~~~
2021-12-24T13:58:14.0863595Z /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/bipedal-locomotion-framework/src/YarpUtilities/autogenerated/src/VectorsCollection.cpp:81:18: note: 'uint32_t _size' previously declared here
2021-12-24T13:58:14.0864976Z    81 |         uint32_t _size;
2021-12-24T13:58:14.0865311Z       |                  ^~~~~
2021-12-24T13:58:14.0867629Z /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/bipedal-locomotion-framework/src/YarpUtilities/autogenerated/src/VectorsCollection.cpp: In member function 'bool BipedalLocomotion::YarpUtilities::VectorsCollection::nested_read_vectors(yarp::os::idl::WireReader&)':
2021-12-24T13:58:14.0870621Z /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/bipedal-locomotion-framework/src/YarpUtilities/autogenerated/src/VectorsCollection.cpp:185:18: error: redeclaration of 'uint32_t _size'
2021-12-24T13:58:14.0871966Z   185 |         uint32_t _size;
2021-12-24T13:58:14.0872317Z       |                  ^~~~~
2021-12-24T13:58:14.0873952Z /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/bipedal-locomotion-framework/src/YarpUtilities/autogenerated/src/VectorsCollection.cpp:167:18: note: 'uint32_t _size' previously declared here
2021-12-24T13:58:14.0875317Z   167 |         uint32_t _size;
2021-12-24T13:58:14.0875664Z       |                  ^~~~~
~~~

**To Reproduce**
I created a test in YARP to easily reproduce the problem: https://github.com/robotology/yarp/pull/2786 . Indeed, if you check CI output you can see:
~~~
~~~

**Expected behavior**
I would expect the compilation to work fine. 

**Configuration (please complete the following information):**
 - OS: Not relevant
 - yarp version: https://github.com/robotology/yarp/commit/7c21bc80ec55255693fe97ad6f10b6e8b730e125
-  compiler: Not relevant

**Additional context**
This was discovered in https://github.com/robotology/robotology-superbuild/pull/971#issuecomment-1000881141 .

The new test added in https://github.com/robotology/yarp/pull/2786 is indeed failing with error:
~~~
[11/56] Building CXX object CMakeFiles/demo_yarp_idl_to_dir.dir/yarp_idl_to_dir/src/MapOfLists.cpp.o
FAILED: CMakeFiles/demo_yarp_idl_to_dir.dir/yarp_idl_to_dir/src/MapOfLists.cpp.o 
/usr/lib/ccache/clang++ -DTHRIFT_INCLUDE_PREFIX -DTHRIFT_NO_NAMESPACE_PREFIX -I/home/runner/work/yarp/yarp/tests/yarpidl_thrift/demo/../../../extern/catch2 -I/home/runner/work/yarp/yarp/build/tests/yarpidl_thrift/demo/yarp_idl_to_dir/include -I/home/runner/work/yarp/yarp/build/tests/yarpidl_thrift/demo -isystem /home/runner/work/yarp/yarp/src/libYARP_os/src -isystem /home/runner/work/yarp/yarp/src/libYARP_conf/src -isystem /home/runner/work/yarp/yarp/build/src/libYARP_conf/src -isystem /home/runner/work/yarp/yarp/src/libYARP_sig/src -fcolor-diagnostics -O3 -DNDEBUG -std=gnu++17 -MD -MT CMakeFiles/demo_yarp_idl_to_dir.dir/yarp_idl_to_dir/src/MapOfLists.cpp.o -MF CMakeFiles/demo_yarp_idl_to_dir.dir/yarp_idl_to_dir/src/MapOfLists.cpp.o.d -o CMakeFiles/demo_yarp_idl_to_dir.dir/yarp_idl_to_dir/src/MapOfLists.cpp.o -c /home/runner/work/yarp/yarp/build/tests/yarpidl_thrift/demo/yarp_idl_to_dir/src/MapOfLists.cpp
yarp_idl_to_dir/src/MapOfLists.cpp:97:18: error: redefinition of '_size'
        uint32_t _size;
                 ^
yarp_idl_to_dir/src/MapOfLists.cpp:79:18: note: previous definition is here
        uint32_t _size;
                 ^
yarp_idl_to_dir/src/MapOfLists.cpp:183:18: error: redefinition of '_size'
        uint32_t _size;
                 ^
yarp_idl_to_dir/src/MapOfLists.cpp:165:18: note: previous definition is here
        uint32_t _size;
                 ^
2 errors generated.
~~~

--- a/src/devices/ILocalization2DMsgs/idl_generated_code/return_get_estimated_poses.cpp
+++ b/src/devices/ILocalization2DMsgs/idl_generated_code/return_get_estimated_poses.cpp
@@ -115,11 +115,11 @@ bool return_get_estimated_poses::read_poses(yarp::os::idl::WireReader& reader)
         return false;
     }
     poses.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    poses.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    poses.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -158,11 +158,11 @@ bool return_get_estimated_poses::nested_read_poses(yarp::os::idl::WireReader& re
         return false;
     }
     poses.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    poses.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    poses.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/IMap2DMsgs/idl_generated_code/return_get_all_areas.cpp
+++ b/src/devices/IMap2DMsgs/idl_generated_code/return_get_all_areas.cpp
@@ -115,11 +115,11 @@ bool return_get_all_areas::read_areas(yarp::os::idl::WireReader& reader)
         return false;
     }
     areas.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    areas.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    areas.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -158,11 +158,11 @@ bool return_get_all_areas::nested_read_areas(yarp::os::idl::WireReader& reader)
         return false;
     }
     areas.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    areas.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    areas.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/IMap2DMsgs/idl_generated_code/return_get_all_locations.cpp
+++ b/src/devices/IMap2DMsgs/idl_generated_code/return_get_all_locations.cpp
@@ -115,11 +115,11 @@ bool return_get_all_locations::read_locations(yarp::os::idl::WireReader& reader)
         return false;
     }
     locations.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    locations.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    locations.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -158,11 +158,11 @@ bool return_get_all_locations::nested_read_locations(yarp::os::idl::WireReader& 
         return false;
     }
     locations.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    locations.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    locations.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/IMap2DMsgs/idl_generated_code/return_get_all_paths.cpp
+++ b/src/devices/IMap2DMsgs/idl_generated_code/return_get_all_paths.cpp
@@ -115,11 +115,11 @@ bool return_get_all_paths::read_paths(yarp::os::idl::WireReader& reader)
         return false;
     }
     paths.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    paths.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    paths.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -158,11 +158,11 @@ bool return_get_all_paths::nested_read_paths(yarp::os::idl::WireReader& reader)
         return false;
     }
     paths.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    paths.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    paths.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/IMap2DMsgs/idl_generated_code/return_get_areas_list.cpp
+++ b/src/devices/IMap2DMsgs/idl_generated_code/return_get_areas_list.cpp
@@ -115,11 +115,11 @@ bool return_get_areas_list::read_areas(yarp::os::idl::WireReader& reader)
         return false;
     }
     areas.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    areas.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    areas.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -158,11 +158,11 @@ bool return_get_areas_list::nested_read_areas(yarp::os::idl::WireReader& reader)
         return false;
     }
     areas.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    areas.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    areas.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/IMap2DMsgs/idl_generated_code/return_get_locations_list.cpp
+++ b/src/devices/IMap2DMsgs/idl_generated_code/return_get_locations_list.cpp
@@ -115,11 +115,11 @@ bool return_get_locations_list::read_locations(yarp::os::idl::WireReader& reader
         return false;
     }
     locations.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    locations.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    locations.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -158,11 +158,11 @@ bool return_get_locations_list::nested_read_locations(yarp::os::idl::WireReader&
         return false;
     }
     locations.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    locations.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    locations.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/IMap2DMsgs/idl_generated_code/return_get_map_names.cpp
+++ b/src/devices/IMap2DMsgs/idl_generated_code/return_get_map_names.cpp
@@ -115,11 +115,11 @@ bool return_get_map_names::read_map_names(yarp::os::idl::WireReader& reader)
         return false;
     }
     map_names.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    map_names.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    map_names.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -158,11 +158,11 @@ bool return_get_map_names::nested_read_map_names(yarp::os::idl::WireReader& read
         return false;
     }
     map_names.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    map_names.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    map_names.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/IMap2DMsgs/idl_generated_code/return_get_paths_list.cpp
+++ b/src/devices/IMap2DMsgs/idl_generated_code/return_get_paths_list.cpp
@@ -115,11 +115,11 @@ bool return_get_paths_list::read_paths(yarp::os::idl::WireReader& reader)
         return false;
     }
     paths.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    paths.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    paths.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -158,11 +158,11 @@ bool return_get_paths_list::nested_read_paths(yarp::os::idl::WireReader& reader)
         return false;
     }
     paths.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    paths.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    paths.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/frameTransformStorageMsgs/idl_generated_code/FrameTransformStorageSetRPC.cpp
+++ b/src/devices/frameTransformStorageMsgs/idl_generated_code/FrameTransformStorageSetRPC.cpp
@@ -368,11 +368,11 @@ bool FrameTransformStorageSetRPC_setTransformsRPC_helper::Command::readArgs(yarp
         return false;
     }
     transforms.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    transforms.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    transforms.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/frameTransformStorageMsgs/idl_generated_code/return_getAllTransforms.cpp
+++ b/src/devices/frameTransformStorageMsgs/idl_generated_code/return_getAllTransforms.cpp
@@ -125,11 +125,11 @@ bool return_getAllTransforms::read_transforms_list(yarp::os::idl::WireReader& re
         return false;
     }
     transforms_list.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    transforms_list.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    transforms_list.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -168,11 +168,11 @@ bool return_getAllTransforms::nested_read_transforms_list(yarp::os::idl::WireRea
         return false;
     }
     transforms_list.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    transforms_list.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    transforms_list.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/multipleAnalogSensorsMsgs/idl_generated_code/SensorMeasurements.cpp
+++ b/src/devices/multipleAnalogSensorsMsgs/idl_generated_code/SensorMeasurements.cpp
@@ -71,11 +71,11 @@ bool SensorMeasurements::read_measurements(yarp::os::idl::WireReader& reader)
         return false;
     }
     measurements.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    measurements.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    measurements.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -114,11 +114,11 @@ bool SensorMeasurements::nested_read_measurements(yarp::os::idl::WireReader& rea
         return false;
     }
     measurements.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    measurements.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    measurements.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/devices/multipleAnalogSensorsMsgs/idl_generated_code/SensorRPCData.cpp
+++ b/src/devices/multipleAnalogSensorsMsgs/idl_generated_code/SensorRPCData.cpp
@@ -143,11 +143,11 @@ bool SensorRPCData::read_ThreeAxisGyroscopes(yarp::os::idl::WireReader& reader)
         return false;
     }
     ThreeAxisGyroscopes.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    ThreeAxisGyroscopes.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    ThreeAxisGyroscopes.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -186,11 +186,11 @@ bool SensorRPCData::nested_read_ThreeAxisGyroscopes(yarp::os::idl::WireReader& r
         return false;
     }
     ThreeAxisGyroscopes.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    ThreeAxisGyroscopes.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    ThreeAxisGyroscopes.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -229,11 +229,11 @@ bool SensorRPCData::read_ThreeAxisLinearAccelerometers(yarp::os::idl::WireReader
         return false;
     }
     ThreeAxisLinearAccelerometers.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    ThreeAxisLinearAccelerometers.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    ThreeAxisLinearAccelerometers.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -272,11 +272,11 @@ bool SensorRPCData::nested_read_ThreeAxisLinearAccelerometers(yarp::os::idl::Wir
         return false;
     }
     ThreeAxisLinearAccelerometers.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    ThreeAxisLinearAccelerometers.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    ThreeAxisLinearAccelerometers.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -315,11 +315,11 @@ bool SensorRPCData::read_ThreeAxisMagnetometers(yarp::os::idl::WireReader& reade
         return false;
     }
     ThreeAxisMagnetometers.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    ThreeAxisMagnetometers.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    ThreeAxisMagnetometers.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -358,11 +358,11 @@ bool SensorRPCData::nested_read_ThreeAxisMagnetometers(yarp::os::idl::WireReader
         return false;
     }
     ThreeAxisMagnetometers.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    ThreeAxisMagnetometers.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    ThreeAxisMagnetometers.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -401,11 +401,11 @@ bool SensorRPCData::read_OrientationSensors(yarp::os::idl::WireReader& reader)
         return false;
     }
     OrientationSensors.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    OrientationSensors.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    OrientationSensors.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -444,11 +444,11 @@ bool SensorRPCData::nested_read_OrientationSensors(yarp::os::idl::WireReader& re
         return false;
     }
     OrientationSensors.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    OrientationSensors.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    OrientationSensors.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -487,11 +487,11 @@ bool SensorRPCData::read_TemperatureSensors(yarp::os::idl::WireReader& reader)
         return false;
     }
     TemperatureSensors.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    TemperatureSensors.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    TemperatureSensors.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -530,11 +530,11 @@ bool SensorRPCData::nested_read_TemperatureSensors(yarp::os::idl::WireReader& re
         return false;
     }
     TemperatureSensors.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    TemperatureSensors.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    TemperatureSensors.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -573,11 +573,11 @@ bool SensorRPCData::read_SixAxisForceTorqueSensors(yarp::os::idl::WireReader& re
         return false;
     }
     SixAxisForceTorqueSensors.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    SixAxisForceTorqueSensors.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    SixAxisForceTorqueSensors.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -616,11 +616,11 @@ bool SensorRPCData::nested_read_SixAxisForceTorqueSensors(yarp::os::idl::WireRea
         return false;
     }
     SixAxisForceTorqueSensors.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    SixAxisForceTorqueSensors.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    SixAxisForceTorqueSensors.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -659,11 +659,11 @@ bool SensorRPCData::read_ContactLoadCellArrays(yarp::os::idl::WireReader& reader
         return false;
     }
     ContactLoadCellArrays.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    ContactLoadCellArrays.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    ContactLoadCellArrays.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -702,11 +702,11 @@ bool SensorRPCData::nested_read_ContactLoadCellArrays(yarp::os::idl::WireReader&
         return false;
     }
     ContactLoadCellArrays.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    ContactLoadCellArrays.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    ContactLoadCellArrays.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -745,11 +745,11 @@ bool SensorRPCData::read_EncoderArrays(yarp::os::idl::WireReader& reader)
         return false;
     }
     EncoderArrays.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    EncoderArrays.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    EncoderArrays.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -788,11 +788,11 @@ bool SensorRPCData::nested_read_EncoderArrays(yarp::os::idl::WireReader& reader)
         return false;
     }
     EncoderArrays.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    EncoderArrays.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    EncoderArrays.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -831,11 +831,11 @@ bool SensorRPCData::read_SkinPatches(yarp::os::idl::WireReader& reader)
         return false;
     }
     SkinPatches.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    SkinPatches.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    SkinPatches.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -874,11 +874,11 @@ bool SensorRPCData::nested_read_SkinPatches(yarp::os::idl::WireReader& reader)
         return false;
     }
     SkinPatches.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    SkinPatches.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    SkinPatches.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -917,11 +917,11 @@ bool SensorRPCData::read_PositionSensors(yarp::os::idl::WireReader& reader)
         return false;
     }
     PositionSensors.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    PositionSensors.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    PositionSensors.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -960,11 +960,11 @@ bool SensorRPCData::nested_read_PositionSensors(yarp::os::idl::WireReader& reade
         return false;
     }
     PositionSensors.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    PositionSensors.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    PositionSensors.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/libYARP_dev/src/idl_generated_code/yarp/dev/Map2DAreaData.cpp
+++ b/src/libYARP_dev/src/idl_generated_code/yarp/dev/Map2DAreaData.cpp
@@ -135,11 +135,11 @@ bool Map2DAreaData::read_points(yarp::os::idl::WireReader& reader)
         return false;
     }
     points.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    points.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    points.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -178,11 +178,11 @@ bool Map2DAreaData::nested_read_points(yarp::os::idl::WireReader& reader)
         return false;
     }
     points.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    points.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    points.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/libYARP_dev/src/idl_generated_code/yarp/dev/Map2DPathData.cpp
+++ b/src/libYARP_dev/src/idl_generated_code/yarp/dev/Map2DPathData.cpp
@@ -81,11 +81,11 @@ bool Map2DPathData::read_waypoints(yarp::os::idl::WireReader& reader)
         return false;
     }
     waypoints.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    waypoints.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    waypoints.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;
@@ -124,11 +124,11 @@ bool Map2DPathData::nested_read_waypoints(yarp::os::idl::WireReader& reader)
         return false;
     }
     waypoints.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    waypoints.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    waypoints.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/yarpdataplayer-console/idl_generated_code/yarpdataplayer_console_IDL.cpp
+++ b/src/yarpdataplayer-console/idl_generated_code/yarpdataplayer_console_IDL.cpp
@@ -2864,11 +2864,11 @@ bool yarpdataplayer_console_IDL_getAllParts_helper::Reply::read(yarp::os::idl::W
         return false;
     }
     return_helper.clear();
-    uint32_t _size;
+    uint32_t _csize;
     yarp::os::idl::WireState _etype;
-    reader.readListBegin(_etype, _size);
-    return_helper.resize(_size);
-    for (size_t _i = 0; _i < _size; ++_i) {
+    reader.readListBegin(_etype, _csize);
+    return_helper.resize(_csize);
+    for (size_t _i = 0; _i < _csize; ++_i) {
         if (reader.noMore()) {
             reader.fail();
             return false;

--- a/src/yarpidl_thrift/src/t_yarp_generator.cc
+++ b/src/yarpidl_thrift/src/t_yarp_generator.cc
@@ -1491,27 +1491,27 @@ void t_yarp_generator::generate_deserialize_container(std::ostringstream& f_cpp_
     bool use_push = tcontainer->has_cpp_name();
 
     f_cpp_ << indent_cpp() << prefix << ".clear();\n";
-    f_cpp_ << indent_cpp() << "uint32_t _size;\n";
+    f_cpp_ << indent_cpp() << "uint32_t _csize;\n";
 
     // Declare variables, read header
     if (ttype->is_map()) {
         // kttpe and vtype available
         f_cpp_ << indent_cpp() << "yarp::os::idl::WireState _ktype;\n";
         f_cpp_ << indent_cpp() << "yarp::os::idl::WireState _vtype;\n";
-        f_cpp_ << indent_cpp() << "reader.readMapBegin(_ktype, _vtype, _size);\n";
+        f_cpp_ << indent_cpp() << "reader.readMapBegin(_ktype, _vtype, _csize);\n";
     } else if (ttype->is_set()) {
         f_cpp_ << indent_cpp() << "yarp::os::idl::WireState _etype;\n";
-        f_cpp_ << indent_cpp() << "reader.readSetBegin(_etype, _size);\n";
+        f_cpp_ << indent_cpp() << "reader.readSetBegin(_etype, _csize);\n";
     } else if (ttype->is_list()) {
         f_cpp_ << indent_cpp() << "yarp::os::idl::WireState _etype;\n";
-        f_cpp_ << indent_cpp() << "reader.readListBegin(_etype, _size);\n";
+        f_cpp_ << indent_cpp() << "reader.readListBegin(_etype, _csize);\n";
         if (!use_push) {
-            f_cpp_ << indent_cpp() << prefix << ".resize(_size);\n";
+            f_cpp_ << indent_cpp() << prefix << ".resize(_csize);\n";
         }
     }
 
     // For loop iterates over elements
-    f_cpp_ << indent_cpp() << "for (size_t _i = 0; _i < _size; ++_i) {\n";
+    f_cpp_ << indent_cpp() << "for (size_t _i = 0; _i < _csize; ++_i) {\n";
     indent_up_cpp();
     {
         if (ttype->is_map()) {
@@ -1541,9 +1541,9 @@ void t_yarp_generator::generate_deserialize_map_element(std::ostringstream& f_cp
 {
     THRIFT_DEBUG_COMMENT(f_cpp_);
 
-    f_cpp_ << indent_cpp() << "uint32_t _size;\n";
+    f_cpp_ << indent_cpp() << "uint32_t _msize;\n";
     f_cpp_ << indent_cpp() << "yarp::os::idl::WireState _lst;\n";
-    f_cpp_ << indent_cpp() << "reader.readListBegin(_lst, _size);\n";
+    f_cpp_ << indent_cpp() << "reader.readListBegin(_lst, _msize);\n";
 
     t_field fkey(tmap->get_key_type(), "_key");
     t_field fval(tmap->get_val_type(), "_val");

--- a/tests/yarpidl_thrift/demo/CMakeLists.txt
+++ b/tests/yarpidl_thrift/demo/CMakeLists.txt
@@ -41,6 +41,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../../../extern/catch2)
 set(IDL_FILES
   annotated.thrift
   demo.thrift
+  mapOfLists.thrift
   namespaced.thrift
   objects3D.thrift
   wrapping.thrift

--- a/tests/yarpidl_thrift/demo/mapOfLists.thrift
+++ b/tests/yarpidl_thrift/demo/mapOfLists.thrift
@@ -1,0 +1,9 @@
+/*
+ * SPDX-FileCopyrightText: 2006-2021 Istituto Italiano di Tecnologia (IIT)
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+struct MapOfLists
+{
+    1: map<string, list<double>> vectors;
+}

